test/connection_test.exs

```diff
@@ -1,6 +1,36 @@
 defmodule Janus.ConnectionTest do
```

Comment:
Try looking at the MockTransport and we shall revisit this, ok?

Reply:
`Janus.MockTransport` lacks in configuration options. For example it doesnt allow to mock a situation when `connect/1` returns `{:error, reason}` or respond to `send/1` with other response then `{:ok, _}`. I suggest to make another issue (or attach its change in this PR) to make those changes. If those changes would be applied I could get rid of `BrokenTransport`.

Reply:
TBH you can just send a janus response that will trigger error. If you return message that follows schema:
```elixir
     %{
           "janus" => "error",
           "transaction" => transaction,
           "error" => %{"code" => code, "reason" => reason}
         }
```

Then the function call will return an error. I don't know whether this satisfies your needs though.

Reply:
Actually not, just using MockTransport i cant reach all cases:

`case transport_module.send(payload_with_transaction, timeout, transport_state) do
      {:ok, new_transport_state} -> ...
      {:error, reason} -> ...
    end`

and I would be missing `{:error, _}` cause `MockTransport` implements `send/1` this way


  `def send(payload, _timeout, %{pairs: pairs} = state) do
    ...
    {:ok, %{state | pairs: pairs}}
  end`

but it is minor issue that is not worth our time